core/src/main/scala/zio/schema/codec/ProtobufCodec.scala

```diff
@@ -0,0 +1,485 @@
+package zio.web.codec
+
+import java.nio.charset.StandardCharsets
+import java.nio.{ ByteBuffer, ByteOrder }
+
+import zio.stream.ZTransducer
+import zio.schema._
+import zio.schema.codec.Codec
+import zio.{ Chunk, ZIO, ZManaged }
+
+// TODO safe splitting of chunks?
+object ProtobufCodec extends Codec {
+  override def encoder[A](schema: Schema[A]): ZTransducer[Any, Nothing, A, Byte] =
+    ZTransducer(ZManaged.succeed(chunk => ZIO.succeed(Encoder.encodeChunk(schema, chunk))))
+
+  override def decoder[A](schema: Schema[A]): ZTransducer[Any, String, Byte, A] =
+    ZTransducer(ZManaged.succeed(chunk => ZIO.fromEither(Decoder.decodeChunk(schema, chunk))))
+
+  object Protobuf {
+
+    sealed trait WireType {}
+
+    case object VarInt                     extends WireType
```

Comment:
I'd put all these in the companion object of `WireType`.

Reply:
Done.